### PR TITLE
Refactor mapper

### PR DESCRIPTION
**Pull request by wcjohnson11**
@sperand-io  @jgershen @ndhoule 

This PR refactors the mapper to return the expected facebook payloads. 

I couldn't include all of the properties needed in the aliases object passed to `msg.properties({aliases})` because of some unexpected behavior for properties named `query`, replicated in [this branch in facade](https://github.com/segmentio/facade/compare/fix/properties-query-alias). [Here's that function in code](https://github.com/segmentio/facade/blob/master/lib/track.js#L161-L178) It seems to grab `query` from the options object, but ignores the `properties` object.

The other issue I faced with is that if you include `quantity` in `msg.properties()` alias dictionary, it will default to 1, even if there is no quantity specified. I would imagine that these are things that we would want to fix in facade but it's possible that there's a reason for this behavior that I'm unaware of.

### CODE REVIEW

**Comment by jgershen** [comment]
looked at `blame`; seems like the facade behavior dates from the dawn of time (2013). @yields, any idea if there's a reason for this?

if no one remembers, let's open a low-pri JIRA task to "figure out why facade does these things and either document the behavior or change it".

LGTM otherwise

**Comment by yields** [comment]
> @yields, any idea if there's a reason for this?

[yup](https://github.com/segmentio/facade/blob/7ac6516d3e4edf51b7d43f0058b10b80afdfe7ef/lib/track.js#L158), it ignores `properties.query` in favor of `options.query`, no clue what `options.query` is, facade does things, lots of things.

> if no one remembers, let's open a low-pri JIRA task to "figure out why facade does these things and either document the behavior or change it".

I think we should get rid of facade wherever possible, i'm in favor of vanilla JS objects passed to the integrations.

So instead of relying on facade to normalize the schema, the integration worker should normalize it and pass a normalized object to the integration instead.

The problem with fixing facade is that it'll likely break somewhere where we depend on `options.query`, or break somewhere else, who knows, it could break an integration a day after the fix, or a year (if we don't update it for a year), lots of issues :(

We talked about doing this since 2013 IIRC, but never got down to actually do it.

**Comment by sperand-io** [comment]
@wcjohnson11 for `track`, are we still checking to see if they are mapping any of their events to fb standard app event names? I'm okay omitting that functionality — if so, we can remove [this line](https://github.com/segment-integrations/integration-facebook-app-events/blob/master/lib/index.js#L23), which currently isn't doing anything afaict

File: lib/mapper.js [hunk 1]
```diff
@@ -212,64 +206,78 @@ function addContext(msg, settings){
  * @param {Object} settings
  */
 
-function customParams(msg, settings){
-  var event = msg.event();
-
+function customParams(msg, settings, eventName){
```
Comment (sperand-io) [inline lib/mapper.js]: is settings used here?

Comment (wcjohnson11) [inline lib/mapper.js]: removed that in latest commit

**Comment by wcjohnson11** [comment]
@sperand-io these latest 2 pr's adds back in support for custom mapping and removes unnecessary settings argument to `customParams` function

**Comment by wcjohnson11** [comment]
@sperand-io Ready for review

**Comment by wcjohnson11** [comment]
@sperand-io This refactors the custom payloads and also adds back in the mapping functionality, this time working.

**Comment by sperand-io** [comment]
🚢

**Pull request merged**